compiler/rustc_mir_build/src/builder/block.rs

```diff
@@ -58,21 +56,21 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         //
         // First we build all the statements in the block.
         let mut let_scope_stack = Vec::with_capacity(8);
-        let outer_source_scope = this.source_scope;
+        let outer_source_scope = self.source_scope;
         // This scope information is kept for breaking out of the parent remainder scope in case
         // one let-else pattern matching fails.
         // By doing so, we can be sure that even temporaries that receive extended lifetime
         // assignments are dropped, too.
         let mut last_remainder_scope = region_scope;
 
-        let source_info = this.source_info(span);
+        let source_info = self.source_info(span);
         for stmt in stmts {
-            let Stmt { ref kind } = this.thir[*stmt];
+            let Stmt { ref kind } = self.thir[*stmt];
             match kind {
                 StmtKind::Expr { scope, expr } => {
-                    this.block_context.push(BlockFrame::Statement { ignores_expr_result: true });
+                    self.block_context.push(BlockFrame::Statement { ignores_expr_result: true });
                     let si = (*scope, source_info);
-                    block = this
+                    block = self
                         .in_scope(si, LintLevel::Inherited, |this| {
                             this.stmt_expr(block, *expr, Some(*scope))
                         })
```

Comment:
See how the code inside this closure uses `this`, but the code outside the closure uses `self`?

Being able to consistently use `this` in both places, regardless of whether they're nested in a closure, is the whole point of doing `let this = self`. So getting rid of it would make the code harder to read and harder to modify.